### DISCUSSION TITLE
Some feedbacks after using it one day.

### DISCUSSION THREAD

**Discussion by xlbao123** [WerWolv/ImHex#1766] [Feedback]
Overall, I am quite satisfied with this tool, but I have encountered some issues and hope someone can help me with them:

1. It crashed twice, both times when I was editing the pattern script. What's worse is that the pattern scripts I wrote were lost. I selected the option to automatically back up the project every 30 seconds in the settings, but it seems that the pattern scripts are not automatically saved.

2. There is too little documentation. Many times, I had to look at other pattern scripts to understand how to implement my own. Eventually, I did get a rough understanding of them. However, there is very little documentation on data processing, and I couldn't find any examples. For instance, I want to know:
- How to process and display a BMP image in data processing.
- Whether it is possible to write a custom node programmatically, such as a PNG decoder node.

**Comment by WerWolv** [comment]
Hey

Auto backups do definitely save the content of the pattern editor. You'll find the backups it created on the welcome screen at the bottom of the "Recent Files" window:
![image](https://github.com/WerWolv/ImHex/assets/10835354/e0951250-227d-4390-841e-d9256bc3867f)

For documentation, I know it's in need to be updated but I simply don't have the time or energy for it while working a full time job.
You can use the `Visualizers -> RGBA8 Image` node to display a raw bitmap image and the `Visualizers -> Image` node to display a PNG. You can write your own nodes by writing a C++ plugin.

**Comment by iTrooz** [comment]
> It crashed twice, both times when I was editing the pattern script.

Hey, could you give me the following:
- ImHex version
- platform
- stacktrace if present (from the logs)

**Comment by paxcut** [comment]
A project file includes all the files that are open and all the patterns that each open file may contain. Saving the project every 30 seconds seems a bit too much but it should work. what happens when you load the auto saved project?

This is an open source free app which is written mostly by one person in his free time. He is making it available so that anybody who wants to contribute can do so be it by fixing bugs or writing docs.Unfortunately not many people are volunteering to write docs so there's nothing we can do about that.

But if you have questions about imhex you can also query the documentation AI by writing it in plain English.in the ask or search box of the online docs.
